torchrl/objectives/dreamer.py

```diff
@@ -148,6 +152,41 @@ def forward(self, tensordict: TensorDict) -> torch.Tensor:
             reward_loss = reward_loss.squeeze(-1)
         reward_loss = reward_loss.mean().unsqueeze(-1)
         # import ipdb; ipdb.set_trace()
+
+        # Alternative:
+        # def forward(self, tensordict: TensorDict) -> Tuple[TensorDict, TensorDict]:
+        #     tensordict = tensordict.copy()
+        #     tensordict.rename_key_(
+        #         ("next", self.tensor_keys.reward),
+        #         ("next", self.tensor_keys.true_reward),
+        #     )
+        #     tensordict = self.world_model(tensordict)
+        #     # compute model loss
+        #     kl_loss = self.kl_loss(
+        #         tensordict.get(("next", self.tensor_keys.prior_mean)),
+        #         tensordict.get(("next", self.tensor_keys.prior_std)),
+        #         tensordict.get(("next", self.tensor_keys.posterior_mean)),
+        #         tensordict.get(("next", self.tensor_keys.posterior_std)),
+        #     )
+        #
+        #     dist: IndependentNormal = self.decoder.get_dist(tensordict)
+        #     reco_loss = -dist.log_prob(
```

Comment:
Going back to distance loss caused the improvements?
I dont mind keeping the distance, just changed it as thats how its done in the original [implementation](https://github.com/danijar/dreamer/blob/master/dreamer.py#L165) but if we keep distance loss we should test both versions or make it selectable?

Reply:
log-prob of a normal is a distance loss if you check the equation. The only difference is (1) we don't add n times log(2*pi) to it and (2) the summing is a bit different here. But in the end they boil down to the same result